pkg/payload/payload.go

```diff
@@ -103,14 +103,23 @@ type Update struct {
 }
 
 func LoadUpdate(dir, releaseImage string) (*Update, error) {
-	payload, tasks, err := loadUpdatePayloadMetadata(dir, releaseImage)
+	klog.V(4).Infof("Loading updatepayload from %q", dir)
+	if err := ValidateDirectory(dir); err != nil {
```

Comment:
is there a case where we want to load from unvalidated directory?? If not, why make each caller to load* validate before calling them..?